### ISSUE TITLE
Failing dotnet database update

### ISSUE THREAD

**Issue by SimonRuehle** [dotnet/efcore#6384]
## Steps to reproduce

Difficult to tell, on some PCs the error is reproducable on others not.
Tested the behaviour with LocalDB and SQL Server Express.
## Expected  behavior

After a dotnet database drop (-f) the database is deleted, if it did not exist nothing changes
After dotnet database update a new one is created.
## Actual behavior

dotnet database drop cannot delete database,
after manual deletion cannot delete because not exists.
Then dotnet database update fails because database exists, even if database drop reports database does not exist.
On file system the database does not exist (in case of LocalDB).
So in short drop fails because not exists, update because exists.
## Environment data

`dotnet --info` output:
.NET Command Line Tools (1.0.0-preview2-003121)

Product Information:
 Version:            1.0.0-preview2-003121
 Commit SHA-1 hash:  1e9d529bc5

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64

**Comment by divega** [comment]
@SimonRuehle it is hard to guess why you couldn't drop the database in the first place but for future reference deleting the mdf/ldf files from disk can leave the SQL Server database in an inconsistent state, e.g. when SQL Server needs to reason about the existence of the database, it normally uses internal system tables stored in the master database rather than the files on disk, so I am not completely surprised that "manual deletion" didn't help.

It would be useful if you could capture and provide the full input and output (with the `-Verbose` flag) of the commands that fail to drop the database.

**Comment by SimonRuehle** [comment]
@divega The point with restoring/resetting LocalDB is valid, if I reset the LocalDB it works again (until the next try to delete the database).
The output:
D:\Projects*\src\*>dotnet ef database drop -f --verbose
Using target project '*'.
Using startup project '*'.
Using framework 'net461'.
Using configuration 'Debug'.
Running C:\Program Files\dotnet\dotnet.exe build D:\Projects\*\src\*\project.json --configuration Debug --
framework net461
Process ID: 7340
Project * (.NETFramework,Version=v4.6.1) was previously compiled. Skipping compilation.
Project * (.NETFramework,Version=v4.6.1) was previously compiled. Skipping compilation.
Project * (.NETFramework,Version=v4.6.1) was previously compiled. Skipping compilation.
Project * (.NETFramework,Version=v4.6.1) was previously compiled. Skipping compilation.
Project * (.NETFramework,Version=v4.6.1) was previously compiled. Skipping compilation.
Setting the data directory to 'D:\Projects\*\src\*\bin\Debug\net461\win7-x64\'.
Invoking dependency command 'Microsoft.EntityFrameworkCore.Design' in project '*'
Running D:\Projects\*\src\*\bin\Debug\net461\win7-x64\Microsoft.EntityFrameworkCore.Design.exe --assembly
D:\Projects\*\src\*\bin\Debug\net461\win7-x64\*.exe --startup-assembly D:\Projects\*\src\Cet
oDocs\bin\Debug\net461\win7-x64\*.exe --dispatcher-version 1.0.0-preview2-21431 --data-dir D:\Projects\*\s
rc\*\bin\Debug\net461\win7-x64\ --project-dir D:\Projects\*\src\* --content-root-path D:\Projects\C
etoDocs\src\* --root-namespace * --verbose database drop -f
Process ID: 2588
Finding DbContext classes...
Using context 'ApplicationDbContext'.
Dropping database '*Angular2'.
Database '*Angular2' did not exist, no action was taken.

D:\Projects\*\src\*>dotnet ef database update --verbose
Using target project '*'.
Using startup project '*'.
Using framework 'net461'.
Using configuration 'Debug'.
Running C:\Program Files\dotnet\dotnet.exe build D:\Projects\*\src\*\project.json --configuration Debug --
framework net461
Process ID: 460
Project WDSEHelp (.NETFramework,Version=v4.6.1) was previously compiled. Skipping compilation.
Project *.DataModel (.NETFramework,Version=v4.6.1) was previously compiled. Skipping compilation.
Project *.Database (.NETFramework,Version=v4.6.1) was previously compiled. Skipping compilation.
Project *.Core (.NETFramework,Version=v4.6.1) was previously compiled. Skipping compilation.
Project * (.NETFramework,Version=v4.6.1) was previously compiled. Skipping compilation.
Setting the data directory to 'D:\Projects\*\src\*\bin\Debug\net461\win7-x64\'.
Invoking dependency command 'Microsoft.EntityFrameworkCore.Design' in project '*'
Running D:\Projects\*\src\*\bin\Debug\net461\win7-x64\Microsoft.EntityFrameworkCore.Design.exe --assembly
D:\Projects\*\src\*\bin\Debug\net461\win7-x64\*.exe --startup-assembly D:\Projects\*\src\Cet
oDocs\bin\Debug\net461\win7-x64\*.exe --dispatcher-version 1.0.0-preview2-21431 --data-dir D:\Projects\*\s
rc\*\bin\Debug\net461\win7-x64\ --project-dir D:\Projects\*\src\* --content-root-path D:\Projects\C
etoDocs\src\* --root-namespace * --verbose database update
Process ID: 2624
Finding DbContext classes...
Using context 'ApplicationDbContext'.
System.Data.SqlClient.SqlException (0x80131904): Cannot create file 'C:\Users\Simon\*Angular2.mdf' because it alr
eady exists. Change the file path or the file name, and retry the operation.
CREATE DATABASE failed. Some file names listed could not be created. Check related errors.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInA
ction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectio
nLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, B
ulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlCommand.RunExecuteNonQueryTds(String methodName, Boolean async, Int32 timeout, Boolean as
yncWrite)
   at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, String methodName, Boo
lean sendToPipe, Int32 timeout, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, String
executeMethod, IReadOnlyDictionary`2 parameterValues, Boolean openConnection, Boolean closeConnection)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteNonQuery(IRelationalConnection connection,
 IReadOnlyDictionary`2 parameterValues, Boolean manageConnection)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.MigrationCommandExecutor.ExecuteNonQuery(IEnumerable`1 migration
Commands, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Storage.Internal.SqlServerDatabaseCreator.Create()
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Migrate(String targetMigration)
   at Microsoft.EntityFrameworkCore.Design.MigrationsOperations.UpdateDatabase(String targetMigration, String contextTyp
e)
   at Microsoft.EntityFrameworkCore.Tools.Cli.DatabaseUpdateCommand.<>c__DisplayClass0_0.<Configure>b__0()
   at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
   at Microsoft.EntityFrameworkCore.Tools.Cli.Program.Main(String[] args)
ClientConnectionId:32168161-df4d-4c55-8624-69cb6b0214ed
Error Number:5170,State:4,Class:16
Cannot create file 'C:\Users\Simon\*Angular2.mdf' because it already exists. Change the file path or the file nam
e, and retry the operation.
CREATE DATABASE failed. Some file names listed could not be created. Check related errors.

**Comment by bricelam** [comment]
LocalDB seems to be in a bad state. Try...

1. Deleting all the `*.mdf` and `*.ldf` files under `%USERPROFILE%`
2. Running the following commands.

        SqlLocalDB stop
        SqlLocalDB delete
3. Trying the EF commands again

**Comment by divega** [comment]
**Triage:** we believe we have answered your question, but feel free to reopen if you still run into issues after applying the recommended measures.
